transactor/src/main/scala/io/mediachain/transactor/Copycat.scala

```diff
@@ -38,15 +40,61 @@ object Copycat {
     }
   }
   
+  class Client(client: CopycatClient) extends JournalClient {
+    private var listeners: Set[JournalListener] = Set()
+    
+    // Journal 
+    def insert(rec: CanonicalRecord): Future[Xor[JournalError, CanonicalEntry]] =
+      FutureConverters.toScala(client.submit(JournalInsert(rec)))
+
+    def update(ref: Reference, cell: ChainCell): Future[Xor[JournalError, ChainEntry]] =
+      FutureConverters.toScala(client.submit(JournalUpdate(ref, cell)))
+    
+    def lookup(ref: Reference): Future[Option[Reference]] = 
+      FutureConverters.toScala(client.submit(JournalLookup(ref)))
+    
+    def currentBlock: Future[JournalBlock] =
+      FutureConverters.toScala(client.submit(JournalCurrentBlock()))
+    
+    // JournalClient
+    def connect(address: String) {
+      client.connect(new Address(address)).join()
+    }
+    
+    def close() {
+      client.close().join()
+    }
+    
+    def listen(listener: JournalListener) {
+      if (listeners.isEmpty) {
```

Comment:
yeah, made the client calls synchronous.
Also, there is a problem with the FutureConverters -- the return CompletbleFuture<Void> which is not convertible to Future[Unit].